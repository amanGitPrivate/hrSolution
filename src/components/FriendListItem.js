import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './FriendListItem.css';

class FriendListItem extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || '',
      gender: this.props.gender || 'noOptionSelected',
      validation: {
        name: true,
        gender: true
      }
    };
    this.starFriend = this.starFriend.bind(this);
    this.deleteFriend = this.deleteFriend.bind(this);
  }

  starFriend() {
    const { id, page, itemsPerPage } = this.props;
    this.props.starFriend(id + (page - 1) * itemsPerPage);
  }

  deleteFriend() {
    const { id, page, itemsPerPage } = this.props;
    this.props.deleteFriend(id + (page - 1) * itemsPerPage);
  }

  render() {
    const { name, starred, gender } = this.props;
    return (
      <li className={styles.friendListItem}>
        <div className={styles.friendInfos}>
          <div>
            <span>{name}</span>
            <i className={`fa fa-${gender}`} />
          </div>
          <div>
            <small>xx friends in common</small>
          </div>
        </div>
        <div className={styles.friendActions}>
          <button
            className={`btn btn-default ${styles.btnAction}`}
            onClick={this.starFriend}
          >
            <i
              className={classnames('fa', {
                'fa-star': starred,
                'fa-star-o': !starred
              })}
            />
          </button>
          <button
            className={`btn btn-default ${styles.btnAction}`}
            onClick={this.deleteFriend}
          >
            <i className="fa fa-trash" />
          </button>
        </div>
      </li>
    );
  }
}

FriendListItem.propTypes = {
  id: PropTypes.number.isRequired,
  page: PropTypes.number.isRequired,
  itemsPerPage: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  starred: PropTypes.bool,
  starFriend: PropTypes.func.isRequired,
  deleteFriend: PropTypes.func.isRequired
};

export default FriendListItem;
