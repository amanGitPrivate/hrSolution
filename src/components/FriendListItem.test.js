import React from 'react';
import FriendListItem from './FriendListItem';
import renderer from 'react-test-renderer';

test('FriendListItem renders correctly', () => {
  const component = renderer.create(
    <FriendListItem
      name={'Aman Seth'}
      id={1}
      starred={false}
      starFriend={jest.fn()}
      deleteFriend={jest.fn()}
      page={2}
      itemsPerPage={2}
      gender={'Male'}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
