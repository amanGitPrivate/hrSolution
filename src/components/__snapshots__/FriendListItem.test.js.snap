exports[`test FriendListItem renders correctly 1`] = `
<li
  className={undefined}>
  <div
    className={undefined}>
    <div>
      <span>
        Aman Seth
      </span>
      <i
        className="fa fa-Male" />
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div
    className={undefined}>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-star-o" />
    </button>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-trash" />
    </button>
  </div>
</li>
`;
